### PR TITLE
Replace X-TRUSTED-HEADER with pub/priv encrypted message

### PR DESCRIPTION
**Pull request by john-westcott-iv**
In simple testing this method appears to be fast but significantly slower than the shared secret method.

Fixes: [AAP-24036](https://issues.redhat.com/browse/AAP-24036)

### CODE REVIEW

**Comment by BrennanPaciorek** [comment]
@john-westcott-iv Do i need to fast-forward this barnch to get the CI tests to run here?

**Comment by BrennanPaciorek** [comment]
There is one remaining line to be covered, though it is covered, since return statements from both code branches are covered. Should we `# NOSONAR` the line?

File: ansible_base/jwt_consumer/common/util.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    try:
+        public_key.verify(
+            bytes.fromhex(signature),
+            bytes(f'{_SHARED_SECRET}-{timestamp}', 'utf-8'),
+            padding.PSS(mgf=padding.MGF1(hashes.SHA256()), salt_length=padding.PSS.MAX_LENGTH),
+            hashes.SHA256(),
+        )
+        return True
+    except InvalidSignature:
+        if not cert.cached:
```
Comment (BrennanPaciorek) [inline ansible_base/jwt_consumer/common/util.py]:
```suggestion
        if not cert.cached: # NOSONAR
```

**Comment by BrennanPaciorek** [comment]
I'll add one more test for a bad cached jwt pubkey, but good not-cached key in `test_util.py`, since that code path is not directly tested. If that does not fix the last line of new code coverage, I will likely `# NOSONAR` the uncovered line.

**Review by djyasin — APPROVED** [review]
Pulled this down and confirmed it works as expected

**Pull request merged**